spack/packages/dla-future-fortran/package.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+# dlaf-no-license-check
+from spack import *
+
+
+class DlaFutureFortran(CMakePackage):
+    """
+    Fortran interface to the DLA-Future library.
+    """
+
+    homepage = "https://github.com/eth-cscs/DLA-Future-Fortran"
+    url      = "http://www.example.com/MyPackage-1.0.tar.gz"
+    git = "https://github.com/eth-cscs/DLA-Future-Fortran"
+
+    maintainers("RMeli")
+    
+    license("BSD-3-Clause")
+
+    version("main")
+
+    variant("shared", default=True, description="Build shared libraries.")
+    variant("test", default=False, description="Build tests.")
+    variant("mpi_f08", default=False, when="+test", description="Build tests.")
+
+    generator("ninja")
+    depends_on("ninja", type="build")
+    depends_on("cmake@3.22:", type="build")
+
+    # Requires 0.3.0 for complex API
+    # Requires 0.4.0 for Intel OneAPI MKL
+    depends_on("dla-future@0.4.0: +scalapack")
+
+    depends_on("mpi", when="+test")
+    depends_on("scalapack", when="+test")
+    depends_on("py-fypp", when="+test", type="build")
+
+    def cmake_args(self):
+        args = []
+
+        args.append(self.define_from_variant("BUILD_SHARED_LIBS", "shared"))
+        args.append(self.define_from_variant("DLAF_FORTRAN_BUILD_TESTING", "test"))
```

Comment:
We have a special `ci-test` variant that enables a bit more than just testing, but that one is not upstream. Otherwise we only have "enable tests on `self.run_tests`": https://github.com/spack/spack/blob/309d3aa1ecf163f219ee0fb5db4d0dd91b28560a/var/spack/repos/builtin/packages/dla-future/package.py#L239.

Reply:
I totally missed this, thanks for pointing it out.

Reply:
As discussed offline, I think that the fact that `fypp` is needed only as `build` dependency for test, prevents the use of `self.run_tests` because marking it as `depends_on("py-fypp", type=("build", "test")` would make it a build dependency of `dla-future-fortran` too (which is not the case).